osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -148,15 +154,49 @@ public void CopyFrom(Mod source)
             if (source.GetType() != GetType())
                 throw new ArgumentException($"Expected mod of type {GetType()}, got {source.GetType()}.", nameof(source));
 
-            foreach (var (_, prop) in this.GetSettingsSourceProperties())
+            foreach (var (_, property) in this.GetSettingsSourceProperties())
             {
-                var targetBindable = (IBindable)prop.GetValue(this)!;
-                var sourceBindable = (IBindable)prop.GetValue(source)!;
+                var targetBindable = (IBindable)property.GetValue(this)!;
+                var sourceBindable = (IBindable)property.GetValue(source)!;
 
                 CopyAdjustedSetting(targetBindable, sourceBindable);
             }
         }
 
+        /// <summary>
+        /// When converting mods from one ruleset to the other, this method makes sure
+        /// to also copy the values of all settings sharing same <see cref="MemberInfo.Name"/> between the two instances.
+        /// </summary>
+        /// <remarks>Copied values are unchanged, even if they have different clamping ranges.</remarks>
+        /// <param name="source">The mod to extract settings from.</param>
+        public void CopyCommonSettings(Mod source)
+        {
+            if (source.UsesDefaultConfiguration)
+                return;
+
+            foreach (var (name, targetSetting) in Settings)
+            {
+                if (!source.Settings.TryGetValue(name, out IBindable? sourceSetting))
+                    continue;
+
+                if (sourceSetting.IsDefault)
+                    continue;
+
+                if (getBindableGenericType(targetSetting) != getBindableGenericType(sourceSetting))
+                    continue;
+
+                // TODO: special case for handling number types
+
+                PropertyInfo property = targetSetting.GetType().GetProperty(nameof(Bindable<bool>.Value))!;
+                property.SetValue(targetSetting, property.GetValue(sourceSetting));
+            }
+
+            Type getBindableGenericType(IBindable setting) =>
+                setting.GetType().GetTypeInheritance()
```

Comment:
This whole change is a bit over my head, so going to leave that to @bdach when we next has a chance to review, but wanted to add that we should definitely not be using this `GetTypeInheritance()` here as it's an automapper internal method.

Reply:
I'll try to check for an alternative later, not really familiar with comparing inherited types.

Reply:
Yes, a much simpler check exists.